### DISCUSSION TITLE
Styling of an element inside editor textbox

### DISCUSSION THREAD

**Discussion by tomas-hruby** [ueberdosis/tiptap#5193] [General Questions]
I need to apply custom styling for an element inside the textbox. My idea is that i would extend the Paragraph extension and give it some custom class:
```
const CustomParagraph = Paragraph.extend({
  HTMLAttributes: {
    class: "file-link",
  },
});
```
Then i would like to use this CustomParagraph inside `onDrop` function in `FileHandler` extension, idealy with `editor.commands.insertContent` command. Then i would use the `file-link` class to style it.
This is what i (unsuccesfully) tried so far:
```
editor.commands.insertContent([
  {
    type: "CustomParagraph",
    content: [
      {
        type: "text",
        text: "File Link",
      },
    ],
  },
]);
```
I will be happy for any advice on how to use the `CustomParagraph`, or any other way of achieving what i described.
Thank you.

**Comment by nperez0111** [comment]
Your CustomParagraph does not change the extension name so it is still only registered as `paragraph` if you add `name: 'customParagraph'` to your extend options you should be able to set it with that name (assuming that you want to keep paragraph & custom paragraph separate). If you don't want to keep them separate, then you can just change `type: "paragraph"` in your insertCotnent

**Comment by tomas-hruby** [comment]
Yes, i want to keep them separate. So i named my extension `customParagraph`. Then i used this `type: "customParagraph"` in `editor.commands.insertContent`, but the element `<p>File Link</p>` still doesn't have `class="file-link"`. Is there somehting else i need to do to apply the class?

**Reply by nperez0111** [reply]
HTMLAttributes is not a top-level property, it is an option so it should be specified like this:

```js
const CustomParagraph = Paragraph.extend(
  name: 'customParagraph',
  addOptions() {
    return {
      ...this.parent?.(),
      HTMLAttributes: {
        class: 'file-link',
      },
    }
  },
})
```

Please read our docs around this for any further questions: https://tiptap.dev/docs/editor/guide/custom-extensions#introduction

**Reply by tomas-hruby** [reply]
Thank you very much.
